x/lockup/keeper/msg_server.go

```diff
@@ -0,0 +1,90 @@
+package keeper
+
+import (
+	"context"
+
+	"github.com/c-osmosis/osmosis/x/gamm/utils"
+	"github.com/c-osmosis/osmosis/x/lockup/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type msgServer struct {
+	keeper Keeper
+}
+
+// NewMsgServerImpl returns an instance of MsgServer
+func NewMsgServerImpl(keeper Keeper) types.MsgServer {
+	return &msgServer{
+		keeper: keeper,
+	}
+}
+
+var _ types.MsgServer = msgServer{}
+
+func (server msgServer) LockTokens(goCtx context.Context, msg *types.MsgLockTokens) (*types.MsgLockTokensResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	lock := types.PeriodLock{
```

Comment:
Can you move this part (the creation of the lock struct) into the keeper function as well?  This will make it easier for other modules to call it if needed.

Reply:
moved